src/racer/ast.rs

```diff
@@ -611,6 +611,28 @@ impl<'c, 's> visit::Visitor for ExprTypeVisitor<'c, 's> {
                 });
             }
 
+            ExprKind::Try(ref expr) => {                
+                debug!("try expr");
+                self.visit_expr(&expr);
+                self.result = if let Some(&Ty::Match(ref m)) = self.result.as_ref() {
+                    // HACK: Try to break open the result and find it's "Ok" type.
+                    // Once the 'Try' operator trait stabilizes, it'd be better to
+                    // find the type through the trait.
+                    if m.matchstr == "Result" && m.generic_types.len() == 2 {
```

Comment:
@jwilm I found a problem with this line; `()` doesn't appear in the `generic_types` list, so `Result<_, ()>` has `generic_types.len() == 1`. Even worse, the same is true of `Result<(), _>`; in that case, `generic_types[0]` gets the `E` case, rather than the `T` case. Is this deliberate, or a bug somewhere else in racer?